### PR TITLE
Remove project auto-reload code, (it is part of CPS now)

### PR DESCRIPTION
**Pull request by jviau**
This PR moves the project reload code into CPS.

`IProjectReloadInterceptor` is now part of the CPS Sdk.

We will need to binary insert CPS and .netcore project system at the same time.

Couple side notes of what I have already fixed with reload code in CPS:

1. evaluation errors no longer crash VS (merged into CPS, not inserted yet)
2. auto-reload when an import changes on disk. (PR is up).
    * Unfortunately, does not handle the case when a conditioned `<Import />` element changes from `false` to `true` due to changes on disk. I will see about doing that in the future.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Reload/ProjectReloadInterceptor.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using System.Collections.Immutable;
 using System.ComponentModel.Composition;
 using Microsoft.Build.Construction;
+using Microsoft.VisualStudio.ProjectSystem.VS;
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Reload/ProjectReloadInterceptor.cs]: The IProjectReloadInterceptor is in your VS namespace? Did it need anything VS specific when you moved it?

**Comment by srivatsn** [comment]
> Unfortunately, does not handle the case when a conditioned <Import /> element changes from false to true due to changes on disk. I will see about doing that in the future

By this you mean if there was something like <Import Condition=Exists(somefile) /> and if somefile wasn't present before but suddenly appears then we'll still require a reload?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Reload/ProjectReloadInterceptor.cs (hunk 1 above)
Comment (jviau) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Reload/ProjectReloadInterceptor.cs]: I will move it to non-VS.

**Comment by jviau** [comment]
It won't need a reload. The import will appear in the next evaluation of the project and will be subscribed to then. But for now I don't automatically trigger the reevaluation when it initially appears.

**Comment by jviau** [comment]
I forgot to put some dev dependencies = true on the package. Going to fix that to fix the build errors.

**Pull request merged**